### ISSUE TITLE
Replace uses of `Locations[0]` and `Locations.First()`

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#29707]
We should replace them with an extension method that handles `Locations` being an empty array.

**Comment by jaredpar** [comment]
Can you clarify this bug a bit? Is this specific to NRT or is it a more general requirement for the compiler?

**Comment by jcouv** [comment]
NRT hit this when we had `NonNullTypesAttribute` (since the injected type didn't have syntax and therefore no location). 
I wouldn't consider urgent/blocking anymore. But it's good hygiene to make the change at some point.
Removed the NRT label

**Comment by jaredpar** [comment]
I agree this is good hygiene to be doing here and we should keep this open as a general issue.
